site/en/docs/configurable-attributes.md

```diff
@@ -999,4 +999,90 @@ Caution: To prevent restarting the Bazel server, invoke `bazel config` with the
 same command line flags as the `bazel cquery`. The `config` command relies on
 the configuration nodes from the still-running server of the previous command.
 
+### Why doesn't select() work with platforms? {:#faq-select-platforms}
+
+Users frequently want to write a configurable attribute that checks whether a
+particular platform is the target platform. Unfortunately, this is not directly
+supported.
+
+The semantics of this are confusing: imagine the following BUILD file:
+
+```py
+platform(
+    name = "x86_linux_platform",
+    constraint_values = [
+        "@platforms//cpu:x86",
+        "@platforms//os:linux",
+    ],
+)
+
+cc_library(
+    name = "lib",
+    srcs = [...],
+    linkopts = select({
+        ":x86_linux_platform": ["--enable_x86_optimizations"],
+        "//conditions:default": [],
+    }),
+)
+```
+
+In this case, which select should be used if the target platform has both the
+`@platforms//cpu:x86` and `@platforms//os:linux` constraints, but is **not** the
+`:x86_linux_platform` defined here? The author of the BUILD file and the user
+who defined the separate platform may have different ideas.
+
+#### What should I do instead?
+
+Instead, you can define a `config_setting` that will match **any** platform with
+these constraints:
+
+```py
+config_setting(
+    name = "is_x86_linux",
+    constraint_values = [
+        "@platforms//cpu:x86",
```

Comment:
I don't know if it's worth mentioning, but I suppose someone could define this constraint list as its own variable, then reference the variable in both the `platform` definition and `config_setting` to guarantee consistency.

Reply:
I thought about that, but felt it was getting too prescriptive.